packages/wallet/src/services/types.ts

```diff
@@ -51,14 +51,17 @@ export interface FailedTx {
   error?: Cardano.TxSubmissionError;
 }
 
+export type ConfirmedNewTxAlonzoWithSlot = Cardano.NewTxAlonzo & { slot: Cardano.PartialBlockHeader['slot'] };
+
 export interface TransactionsTracker {
   readonly history$: Observable<Cardano.TxAlonzo[]>;
+  readonly rollback$: Observable<Cardano.TxAlonzo[]>;
```

Comment:
I think an array here is confusing. We only need 1 notification per rollback since we're setting up the resubmitter on startup.

```suggestion
  readonly rollback$: Observable<Cardano.TxAlonzo>;
```

Reply:
I think there could be multiple rolled back transactions detected in the same [tipHeight iteration](https://github.com/input-output-hk/cardano-js-sdk/pull/351/files#diff-b015172e50668407821a584b9a03f1c2c70ccf60bd27d3832335c85360d86d96).
But I can turn it into a stream before processing.

Reply:
Yep there can be, just emit them all one by one then

Reply:
Updated in [0286cb7](https://github.com/input-output-hk/cardano-js-sdk/pull/351/commits/0286cb7b5c3844765304acc03f3163345df65fa2)